v1/src/test/java/com/google/cloud/teleport/spanner/AvroRecordConverterTest.java

```diff
@@ -258,6 +258,103 @@ public void testParseFloat() {
     assertThrows(IllegalArgumentException.class, () -> avroRecordConverter.apply(avroRecord1));
   }
 
+  @Test
+  public void float32Array() {
+    String colName = "arrayoffloat32";
+    Schema schema = createArrayAvroSchema(colName, FLOAT);
+
+    // Null field
+    GenericRecord avroRecord = new GenericRecordBuilder(schema).set("id", 0).build();
+    Optional<List<Float>> result = AvroRecordConverter.readFloat32Array(avroRecord, FLOAT, colName);
+    assertFalse(result.isPresent());
+
+    // Convert from float to Float32.
+    avroRecord = new GenericRecordBuilder(schema).set("id", 0).set(colName, floatArray).build();
+    result = AvroRecordConverter.readFloat32Array(avroRecord, FLOAT, colName);
+    assertArrayEquals(floatArray.toArray(), result.get().toArray());
+
+    // Convert from int to Float32.
+    avroRecord = new GenericRecordBuilder(schema).set("id", 0).set(colName, intArray).build();
+    result = AvroRecordConverter.readFloat32Array(avroRecord, INT, colName);
+    assertArrayEquals(floatArray.toArray(), result.get().toArray());
+
+    // Convert from String to Float32.
+    avroRecord = new GenericRecordBuilder(schema).set("id", 0).set(colName, stringArray).build();
+    result = AvroRecordConverter.readFloat32Array(avroRecord, STRING, colName);
+    assertArrayEquals(floatArray.toArray(), result.get().toArray());
```

Comment:
Can we also cover what happens if string cast is not possible?

Reply:
Done